src/mon/ElectionLogic.cc

```diff
@@ -214,10 +262,28 @@ void ElectionLogic::receive_ack(int from, epoch_t from_epoch)
   }
 }
 
+bool ElectionLogic::victory_makes_sense(int from)
+{
+  bool makes_sense = false;
+  switch(strategy) {
+  case CLASSIC:
+    makes_sense = (from < elector->get_my_rank());
```

Comment:
might want to add `break` here? otherwise `CLASSIC` would be an alias of `DISALLOW`.